### PR TITLE
CI: migrate pipeline into GH Actions

### PR DESCRIPTION
**Pull request by jportasa**
- [x] Add version print logic into codebase 
- [x] Add docker login for pull rate limitations 
- [x] Add snyk step as `needs` for prerelease once it will be ok
- [x] Migrate into Org secrets
- [x] Change upgrade codes for windows (keep old codes)
- [x] Squash commits because Joan Porta did not accept CLA
- [x] Org slack secrets

### CODE REVIEW

File: build/package/windows/nri-amd64-installer/Product.wxs [hunk 1]
```diff
@@ -67,29 +67,32 @@
 
     <Fragment>
         <ComponentGroup Id="CustomPluginsBinComponent" Directory="CustomPluginsBinFolder">
-            <Component Id="CMP_NRI_{IntegrationName}_EXE" Guid="eea30c6e-7a14-4cc5-97c5-88a742041643" Win64="yes">
-                <File Id="FILE_NRI_{IntegrationName}_EXE"
-                      Source="$(var.BinariesPath){IntegrationExe}"
+            <Component Id="CMP_NRI_$(var.IntegrationName)_EXE" Guid="eea30c6e-7a14-4cc5-97c5-88a742041643" Win64="yes">
```
Comment (paologallinaharbur) [inline build/package/windows/nri-amd64-installer/Product.wxs:70]: I am not sure why we are changing all Guid, it should not be an issue but I am not 100% sure

**Review by paologallinaharbur — APPROVED** [review]
lgtm

File: build/package/windows/nri-amd64-installer/Product.wxs (hunk 1 above)
Comment (roobre) [reply build/package/windows/nri-amd64-installer/Product.wxs:70]: I don't remember making this change so I guess this was done before we took over. I'm not that knowledgeable of windows "package management" so I don't really know the implications of this. Maybe someone can clarify whether this is desired or needed somewhere else? @brushknight @ardias @gsanchezgavier

Comment (paologallinaharbur) [reply build/package/windows/nri-amd64-installer/Product.wxs:70]: also @davidbrota

Comment (roobre) [reply build/package/windows/nri-amd64-installer/Product.wxs:70]: We decided to keep them as they were in `master` reverting previous changes if necessary.

**Comment by roobre** [comment]
No GUID changes found, ready to merge.

**Pull request merged**